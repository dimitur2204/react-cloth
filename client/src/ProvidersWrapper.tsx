import React, { PropsWithChildren } from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter } from 'react-router-dom'

export interface ProvidersWrapperProps {
  store: any
}

const ProvidersWrapper = ({ children, store }: PropsWithChildren<ProvidersWrapperProps>) => (
  <React.StrictMode>
    <Provider store={store}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  </React.StrictMode>
)

export default ProvidersWrapper
