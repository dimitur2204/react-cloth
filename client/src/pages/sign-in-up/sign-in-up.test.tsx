import { shallow, ShallowWrapper } from 'enzyme';
import configureStore from 'redux-mock-store';

import SignInUp from './sign-in-up.component';
import ProvidersWrapper from '../../ProvidersWrapper';

describe('SignInUp', () => {
	let wrapper: ShallowWrapper;

	beforeEach(() => {
		const mockStore = configureStore();
		const store = mockStore({});
		wrapper = shallow(
			<ProvidersWrapper store={store}>
				<SignInUp />
			</ProvidersWrapper>
		);
	});

	it('should render SignInUp component', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
