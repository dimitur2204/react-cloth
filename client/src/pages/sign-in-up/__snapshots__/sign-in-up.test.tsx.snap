// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInUp should render SignInUp component 1`] = `ShallowWrapper {}`;
